lua/fpp/server/ownability.lua

```diff
@@ -267,51 +267,58 @@ end
 Networking
 ---------------------------------------------------------------------------]]
 util.AddNetworkString("FPP_TouchabilityData")
--- Sends 32 + 32 + 5 + 20 = 89 bits of ownership data per entity
+-- Sends 13 + 8 + 5 + 20 = 46 bits of ownership data per entity
 local function netWriteEntData(ply, ent)
-    -- EntIndex for when it's out of the PVS of the player
-    net.WriteUInt(ent:EntIndex(), 32)
+    -- EntIndex for when it's out of the PVS of the player`
+    net.WriteUInt(ent:EntIndex(), 13)
 
     local owner = ent:CPPIGetOwner()
-    net.WriteUInt(IsValid(owner) and owner:EntIndex() or -1, 32)
+    net.WriteUInt(IsValid(owner) and owner:EntIndex() or -1, 8)
     net.WriteUInt(ent.FPPRestrictConstraint and ent.FPPRestrictConstraint[ply] or ent.FPPCanTouch[ply], 5) -- touchability information
     net.WriteUInt(ent.FPPConstraintReasons and ent.FPPConstraintReasons[ply] or ent.FPPCanTouchWhy[ply], 20) -- reasons
 end
 
-function FPP.plySendTouchData(ply, ents)
-    local count = #ents
-
-    if count == 0 then return end
+local touchabilityDataQueue = {}
+local function startNetWriteQueue()
+    if timer.Exists( "FPP_TouchabilityQueue" ) then return end
+
+    timer.Create( "FPP_TouchabilityQueue", 0, 0, function()
+        for ply, entities in pairs(touchabilityDataQueue) do
+            if not IsValid(ply) then touchabilityDataQueue[ply] = nil continue end
+
+            local count = 0
+            net.Start("FPP_TouchabilityData")
+            for ent in pairs(entities) do
+                netWriteEntData(ply, ent)
+                count = count + 1
+                touchabilityDataQueue[ply][ent] = nil
+                if count >= 1400 then break end
```

Comment:
I'm not sure I understand this number. With your change, the message is smaller, but the limit is smaller. Where does the number 1400 come from?

Reply:
It's in the PR description
"The reason i lowered the count per message from 5825 to 1400 is because we're sending 46bits while the max net message size is 65533bits so 65533÷46."
I feel like gmod internally compresses it in some way but im not entirely sure, i figured 1400 ents/s should be fast enough as the pr makes it half the amount it took before anyways